glib/src/subclass/signal.rs

```diff
@@ -377,7 +378,24 @@ enum SignalRegistration {
     },
 }
 
-impl<'a> SignalBuilder<'a> {
+impl SignalBuilder {
+    // rustdoc-stripper-ignore-next
+    /// The signal parameters.
+    pub fn param_types(mut self, param_types: &[Type]) -> Self {
+        self.param_types = param_types
+            .iter()
+            .map(|t| SignalType::from(*t))
+            .collect::<Vec<_>>();
+        self
+    }
+
+    // rustdoc-stripper-ignore-next
+    /// The signal is returned value type.
+    pub fn return_type<T: StaticType>(mut self) -> Self {
```

Comment:
This doesn't allow for marking the type `G_SIGNAL_TYPE_STATIC_SCOPE` anymore

Reply:
What do  you suggest we do here ? Revert the change or add another variant that automatically wraps things with `SignalType::with_static_type`?

Reply:
I would here do

```fn
fn param_types(mut self, param_types: impl IntoIterator<Item = Into<SignalType>>) -> Self;
fn return_type(mut self, type_: impl Into<SignalType>) -> Self;
```

The latter might also be in addition, having something like `fn return_type_with_type()` (but a better name please :P )

Reply:
Also IMHO having all functions take an actual `type` parameter is fine instead of using `T: StaticType`. The latter doesn't bring much improvement

Reply:
> Also IMHO having all functions take an actual type parameter is fine instead of using T: StaticType. The latter doesn't bring much improvement

It avoid having to write `T::static_type()` everywhere but yeah that can work. I will submit a patch later today

Reply:
But how often do you create signals with non-unit return types? :)

Reply:
For gui apps, a lot of times :)

Reply:
Oh really? I thought that's just something people do in GStreamer often